core/services/ocrcommon/transmitter.go

```diff
@@ -65,6 +68,39 @@ func NewTransmitter(
 	}, nil
 }
 
+func NewTransmitterWithStatusChecker(
+	txm txManager,
+	fromAddresses []common.Address,
+	gasLimit uint64,
+	effectiveTransmitterAddress common.Address,
+	strategy types.TxStrategy,
+	checker txmgr.TransmitCheckerSpec,
+	chainID *big.Int,
+	keystore roundRobinKeystore,
+	statuschecker statuschecker.TransactionStatusChecker,
```

Comment:
We are instantiating the transmitter in the `evm.go` file which is CCIP. Having it in `ccip` would mean forking the evm.go file too, right?

Reply:
The ccip repo is a fork of the entire chainlink repo, so evm.go is there as well

Reply:
Shouldn't we refrain from changing the common files (= files susceptible to being changed from `chainlink`)? It will simplify a lot my changes, but in the long run, doing so might make resolving conflicts worse when merging from `chainlink` to `ccip`

Reply:
No, I'm not proposing changing the common component, but creating a new implementation that is mostly copy-paste the existing one with the status checker added.

Its actually more likely this gets removed in the future because it will probably see no uses in core as its CCIP specific. This has happened quite a lot of times because the CCIP usages are not checked when developers are changing core.

Reply:
Moved the logic in `ccip` repo by creating a `newCCIPOnChainContractTransmitter` in the ccip.go file (which is part of `ccip`.